### PR TITLE
fix: Added currency symbols to loan account summary

### LINKED ISSUE #1112: Add currency symbol in Account Summary
**Summary:** 

Need to add currency symbol with the amount whenever charges or balance are mentioned in the account summary. This would make it easier for the user and cause less confusion.

**Steps to reproduce:** 

Open a client, expand any account (loan or savings) and click on 
it to get to account summary.

**Screenshots:** 
Screenshot_20190309-214235

### PR DESCRIPTION
**Pull request by PiyushSharma99**
Fixes #1112 

Please make sure these boxes are checked before submitting your pull request - thanks!

- [x] Apply the `MifosStyle.xml` style template to your code in Android Studio.

- [x] Run the unit tests with `./gradlew check` to make sure you didn't break anything

- [x] If you have multiple commits please combine them into one commit by squashing them.

### CODE REVIEW

**Comment by PiyushSharma99** [comment]
@santoshconflux @jddeep Please review and approve. Thanks.

**Comment by laxyapahuja** [comment]
@PiyushSharma99 Don't you think that adding currency symbols in front of every amount would be better?

**Comment by laxyapahuja** [comment]
Also, making a new view just to add one currency symbol doesn't seem convenient and takes a lot of lines of code. I'd suggest working on a workaround. This can be done in a lot less lines :)

**Comment by PiyushSharma99** [comment]
Okay @laxyapahuja Sure I will try to think of more ideas and make the changes as soon as I can. Thanks.

**Comment by luckyman20** [comment]
@PiyushSharma99 Status? Will wait for another 3 days for you to respond otherwise PR would be closed.

**Comment by PiyushSharma99** [comment]
Done @luckyman20

**Comment by laxyapahuja** [comment]
> Done @luckyman20

hey. the current fix that you've done doesn't place a '$' before every amount. it only adds one in the end.

**Comment by PiyushSharma99** [comment]
> > Done @luckyman20
> 
> hey. the current fix that you've done doesn't place a '$' before every amount. it only adds one in the end.

Hi, you might wanna check again :)
Screenshot_2019-12-25-15-06-37-25_3b69eb5d9a470d17be550edd2848b555

**Comment by laxyapahuja** [comment]
> Also, making a new view just to add one currency symbol doesn't seem convenient and takes a lot of lines of code. I'd suggest working on a workaround. This can be done in a lot less lines :)

"This can be done in a lot less lines."

**Comment by PiyushSharma99** [comment]
> "This can be done in a lot less lines."

Hi! Many things can be done in pot less lines. But I have made the PR with the issue entirely resolved. If you have some better idea please pitch it directly. Will see if I can do the change at this stage as I have already solved the issue. Thanks.

**Comment by laxyapahuja** [comment]
> > "This can be done in a lot less lines."
> 
> Hi! Many things can be done in a lot less lines. But I have made the PR with the issue entirely resolved. If you have some better idea please pitch it directly. Will see if I can do the change at this stage as I have already solved the issue. Thanks
> 
> > hey. the current fix that you've done doesn't place a '$' before every amount. it only adds one in the end.
> 
> Also please check before commenting :)

Still can't see one before "0.0" in "In Arrears" :D

**Comment by PiyushSharma99** [comment]
> Still can't see one before "0.0" in "In Arrears" :D

Lol Added that. :D

File: mifosng-android/src/main/res/layout/fragment_loan_account_summary.xml [hunk 1]
```diff
@@ -109,20 +109,29 @@
...
                 android:text="@string/loan_amount_disbursed"
                 android:textAppearance="?android:attr/textAppearanceMedium" />
 
+            <TextView
+                android:layout_width="wrap_content"
+                android:layout_height="wrap_content"
+                android:layout_below="@+id/divider_2"
+                android:layout_marginTop="8dp"
+                android:layout_toLeftOf="@+id/tv_amount_disbursed"
+                android:text="$ "
```
[the remark this answers is not in the document]
Comment (PiyushSharma99) [reply mifosng-android/src/main/res/layout/fragment_loan_account_summary.xml]: @luckyman20 Can you please guide me on how to do that?

File: mifosng-android/src/main/res/layout/fragment_loan_account_summary.xml [hunk 2]
```diff
@@ -490,8 +690,8 @@
                 android:layout_width="match_parent"
                 android:layout_height="wrap_content"
                 android:layout_below="@+id/tbl_loanSummary"
-                android:layout_marginBottom="8dp"
                 android:layout_marginTop="8dp"
+                android:layout_marginBottom="8dp"
```
[the remark this answers is not in the document]
Comment (PiyushSharma99) [reply mifosng-android/src/main/res/layout/fragment_loan_account_summary.xml]: Sure, will remove

File: mifosng-android/src/main/res/layout/fragment_loan_account_summary.xml [hunk 3]
```diff
@@ -109,20 +109,29 @@
...
+                android:layout_marginTop="8dp"
+                android:layout_toLeftOf="@+id/tv_amount_disbursed"
+                android:text="$ "
+                android:textAppearance="?android:attr/textAppearanceMedium" />
+
             <TextView
                 android:id="@+id/tv_amount_disbursed"
                 android:layout_width="wrap_content"
                 android:layout_height="wrap_content"
+                android:layout_below="@+id/divider_2"
```
[the remark this answers is not in the document]
Comment (PiyushSharma99) [reply mifosng-android/src/main/res/layout/fragment_loan_account_summary.xml]: @luckyman20 If I removed this the positioning of the view was undesirable :thinking:

File: mifosng-android/src/main/res/layout/fragment_loan_account_summary.xml (hunk 2 above)
[the remark this answers is not in the document]
Comment (PiyushSharma99) [reply mifosng-android/src/main/res/layout/fragment_loan_account_summary.xml]: @luckyman20 Thank you so much for the review. With some guidance, I believe I can fix the PR for merging. Thank you.

**Comment by PiyushSharma99** [comment]
@luckyman20 Thank you so much for the review. With some guidance, I believe I can fix the PR for merging. Thank you.

File: mifosng-android/src/main/res/layout/fragment_loan_account_summary.xml (hunk 1 above)
[the remark this answers is not in the document]
Comment (luckyman20) [reply mifosng-android/src/main/res/layout/fragment_loan_account_summary.xml]: Check how to fetch and use API response.

File: mifosng-android/src/main/res/layout/fragment_loan_account_summary.xml (hunk 3 above)
[the remark this answers is not in the document]
Comment (luckyman20) [reply mifosng-android/src/main/res/layout/fragment_loan_account_summary.xml]: You have just changed it place. I don't think that makes any difference. Does it?

[the remark this answers is not in the document]
Comment (PiyushSharma99) [reply mifosng-android/src/main/res/layout/fragment_loan_account_summary.xml]: Oh! I see in that sense. Cool will fix this.

**Comment by laxyapahuja** [comment]
@PiyushSharma99 are you still working on this?

**Comment by luckyman20** [comment]
@PiyushSharma99 It will be closed in one day if no activity shown.

**Comment by laxyapahuja** [comment]
Can you upload an updated screenshot?

File: mifosng-android/src/main/java/com/mifos/mifosxdroid/online/loanaccountsummary/LoanAccountSummaryFragment.java [hunk 4]
```diff
@@ -169,6 +177,27 @@ public static LoanAccountSummaryFragment newInstance(int loanAccountNumber,
...
 
+    static {
+        currencyLocaleMap = new TreeMap<Currency, Locale>(new Comparator<Currency>() {
+            public int compare(Currency c1, Currency c2) {
+                return c1.getCurrencyCode().compareTo(c2.getCurrencyCode());
+            }
+        });
+        for (Locale locale : Locale.getAvailableLocales()) {
+            try {
+                Currency currency = Currency.getInstance(locale);
```
Comment (luckyman20) [inline mifosng-android/src/main/java/com/mifos/mifosxdroid/online/loanaccountsummary/LoanAccountSummaryFragment.java]: You have to fetch from the API response, not by getting `Locale` of the device.

**Comment by luckyman20** [comment]
Closing this because of inactivity.

**Pull request closed without merging**